atlasdb-client/src/main/java/com/palantir/atlasdb/util/TaggedMetricsInvocationEventHandler.java

```diff
@@ -151,7 +151,7 @@ private Timer getSuccessTimerExtraTags(MethodWithExtraTags methodWithExtraTags)
         return extraTagsTimerCache.computeIfAbsent(methodWithExtraTags, onSuccessTimerMappingFunctionExtraTags);
     }
 
-    @Value.Immutable
+    @Value.Immutable(lazyhash = true)
     interface MethodWithExtraTags {
```

Comment:
this type is used as a key in `ConcurrentMap<MethodWithExtraTags, Timer> extraTagsTimerCache` so would benefit from memoized hashCode to avoid recomputing the full extraTags Map hashCode on each access.

https://github.com/palantir/atlasdb/blob/31f015c747daffc57499dad665ba0c56a0f60596/atlasdb-client/src/main/java/com/palantir/atlasdb/util/TaggedMetricsInvocationEventHandler.java#L67